airbyte-workers/src/main/java/io/airbyte/workers/helper/FailureHelper.java

```diff
@@ -95,6 +95,18 @@ public static AttemptFailureSummary failureSummaryForCancellation(final Long job
     return failureSummary(failures, partialSuccess);
   }
 
+  public static AttemptFailureSummary failureSummaryForTemporalCleaningJobState(final Long jobId, final Integer attemptNumber) {
+    final FailureReason failureReason = new FailureReason()
+        .withFailureOrigin(FailureOrigin.AIRBYTE_PLATFORM)
+        .withFailureType(FailureType.SYSTEM_ERROR)
+        .withInternalMessage(
+            "Setting attempt to FAILED because the temporal workflow for this connection was restarted, and existing job state was cleaned.")
+        .withExternalMessage("Airbyte internal connection workflow was restarted, causing incomplete jobs to fail. No user action required.")
```

Comment:
```suggestion
        .withExternalMessage("An internal Airbyte error has occurred. Airbyte will attempt to run this job again.")
```

Reply:
@andyjih So I actually went back on this decision a bit - I think it can be misleading in some cases to say that we will run this job again, for example if the connection is set up to be a manual-sync connection (i.e. no automated sync schedule), then Airbyte will _not_ automatically attempt the job again

Reply:
This was the new copy that I settled on based on that: https://github.com/airbytehq/airbyte/pull/12589/files#diff-d7017da9325e432674b6697a0c7ea8eb3b59cdada7d129b29ade4d7c19bd57e8R104
```
"An internal Airbyte connection restart caused this job to fail. No user action required."
```
What do you think?

Reply:
Maybe it is also misleading to say that no user action is required in the manual sync connection case?

Reply:
Ah, good point. Any way to conditionally show different error messages based on whether the connection is scheduled or manual? Or is the temporal logic pretty removed from that?

Reply:
The temporal logic where we are cleaning the job state is fairly removed from that. We could make it work if absolutely necessary, but it would involve adding more queries and activity methods which feels sort of heavy-handed for deciding which error message to show. 

Since we don't expect this to happen often, I would lean toward just picking an error message that is adequate for both cases

Reply:
Yeah, agreed that it's not worth it to add that complexity especially when it's just for an error. 
 
We could make it more verbose and specific:

"An internal Airbyte connection restart caused this job to fail. Connections with schedules will be retried automatically. Manual connections will need to be started again."

Or we can ask them to try syncing again regardless of if they have a schedule or not, since there is the Sync Now button.
"An internal Airbyte error has occurred. This sync will need to be retried."

wdyt?

Reply:
Both good suggestions! I think I like the second option, since it notes what needs to be done, whereas the first option is sort of re-explaining how manual and scheduled connections work which doesn't feel prudent to do in an error message. Thanks for your input!